src/createExplorer.ts

```diff
@@ -80,83 +80,90 @@ class Explorer {
-      },
-    );
-  }
+    const startDirectory = await getDirectory(searchFrom);
+    const result = await this.searchFromDirectory(startDirectory);
 
+    return result;
+  }
 
   public searchSync(searchFrom: string = process.cwd()): CosmiconfigResult {
-    const dir = getDirectorySync(searchFrom);
-    return this.searchFromDirectorySync(dir);
+    const startDirectory = getDirectorySync(searchFrom);
+    const result = this.searchFromDirectorySync(startDirectory);
+
+    return result;
   }
 
   private async searchFromDirectory(dir: string): Promise<CosmiconfigResult> {
     const absoluteDir = path.resolve(process.cwd(), dir);
+
     const run = async (): Promise<CosmiconfigResult> => {
-      return this.searchDirectory(absoluteDir).then(
-        async (result): Promise<CosmiconfigResult> => {
-          const nextDir = this.nextDirectoryToSearch(absoluteDir, result);
-          if (nextDir) {
-            return this.searchFromDirectory(nextDir);
-          }
-          return this.config.transform(result);
-        },
-      );
+      const result = await this.searchDirectory(absoluteDir);
+      const nextDir = this.nextDirectoryToSearch(absoluteDir, result);
+
+      if (nextDir) {
+        return this.searchFromDirectory(nextDir);
+      }
+
+      const transformResult = await this.config.transform(result);
+
+      return transformResult;
     };
 
     if (this.searchCache) {
       return cacheWrapper(this.searchCache, absoluteDir, run);
     }
+
     return run();
   }
 
   private searchFromDirectorySync(dir: string): CosmiconfigResult {
     const absoluteDir = path.resolve(process.cwd(), dir);
+
     const run = (): CosmiconfigResult => {
       const result = this.searchDirectorySync(absoluteDir);
       const nextDir = this.nextDirectoryToSearch(absoluteDir, result);
+
       if (nextDir) {
         return this.searchFromDirectorySync(nextDir);
       }
+
+      const transformResult = this.config.transform(result);
+
       // @ts-ignore
-      return this.config.transform(result);
+      return transformResult;
     };
 
     if (this.searchSyncCache) {
       return cacheWrapperSync(this.searchSyncCache, absoluteDir, run);
     }
+
     return run();
   }
 
   private async searchDirectory(dir: string): Promise<CosmiconfigResult> {
-    return this.config.searchPlaces.reduce(
-      async (
-        prevResultPromise: Promise<CosmiconfigResult>,
-        place,
-      ): Promise<CosmiconfigResult> => {
-        return prevResultPromise.then(
-          async (prevResult: CosmiconfigResult): Promise<CosmiconfigResult> => {
-            if (this.shouldSearchStopWithResult(prevResult)) {
-              return prevResult;
-            }
-            return this.loadSearchPlace(dir, place);
-          },
-        );
-      },
-      Promise.resolve(null),
-    );
+    for await (const place of this.config.searchPlaces) {
+      const tryToLoadConfig = await this.loadSearchPlace(dir, place);
```

Comment:
How about naming this variable `placeResult` instead? The fact that `tryToLoadConfig` starts with a verb makes it seem like a function name.

Reply:
Updated.